include/aws/common/process.h

```diff
@@ -21,6 +21,54 @@ AWS_EXTERN_C_BEGIN
 
 AWS_COMMON_API int aws_get_pid(void);
 
+struct aws_run_command_result {
+    struct aws_allocator *allocator;
+
+    /* error code when calling aws_run_command() */
+    int error_code;
```

Comment:
don't have the AWS_ERROR code be part of the result.
The error-reporting pattern for functions that return a pointer is:
if NULL is returned then check `aws_last_error()` to find out what went wrong.

that means you'd need to be sure `aws_raise_error(XXXX)` is called in your function before NULL is returned

Reply:
Good idea.!